1.Make an API request that returns the months where "fire" was searched in 2016. Which month had the most searches? 

In [8]:
import requests
import pandas as pd
import matplotlib.pyplot as plt

%matplotlib inline

In [93]:
endpoint = 'https://data.nashville.gov/resource/fuaa-r5cm.json?$limit=48000'

In [94]:
params = {
    'year': '2016',
    'query_text': 'fire'
}

In [95]:
response = requests.get(endpoint, params = params)

In [96]:
response

<Response [200]>

In [97]:
result = response.json()

In [98]:
result

[{'month_name': 'January',
  'year': '2016',
  'query_count': '19',
  'query_text': 'fire'},
 {'month_name': 'February',
  'year': '2016',
  'query_count': '35',
  'query_text': 'fire'},
 {'month_name': 'March',
  'year': '2016',
  'query_count': '32',
  'query_text': 'fire'},
 {'month_name': 'April',
  'year': '2016',
  'query_count': '26',
  'query_text': 'fire'},
 {'month_name': 'May',
  'year': '2016',
  'query_count': '24',
  'query_text': 'fire'},
 {'month_name': 'June',
  'year': '2016',
  'query_count': '31',
  'query_text': 'fire'},
 {'month_name': 'July',
  'year': '2016',
  'query_count': '24',
  'query_text': 'fire'},
 {'month_name': 'August',
  'year': '2016',
  'query_count': '47',
  'query_text': 'fire'},
 {'month_name': 'September',
  'year': '2016',
  'query_count': '36',
  'query_text': 'fire'},
 {'month_name': 'October',
  'year': '2016',
  'query_count': '38',
  'query_text': 'fire'},
 {'month_name': 'November',
  'year': '2016',
  'query_count': '32',
  'query_text

In [99]:
fire = pd.DataFrame(result)
print (fire)

   month_name  year query_count query_text
0     January  2016          19       fire
1    February  2016          35       fire
2       March  2016          32       fire
3       April  2016          26       fire
4         May  2016          24       fire
5        June  2016          31       fire
6        July  2016          24       fire
7      August  2016          47       fire
8   September  2016          36       fire
9     October  2016          38       fire
10   November  2016          32       fire
11   December  2016          38       fire


In [100]:
Max_count =(fire['query_count'].max()) 

In [101]:
print(Max_count)

47


In [102]:
max_fire=fire[fire['query_count'] == fire['query_count'].max()]
max_fire

month_name  year query_count query_text
7     August  2016          47       fire

In [103]:
2.Make an API request that returns all the times a query was run more than 100 times in a month. How many times did this occur?

Object `occur` not found.


In [130]:
endpoint1 = 'https://data.nashville.gov/resource/fuaa-r5cm.json?$where=query_count > 100 &$limit=10000'

In [131]:
response1 = requests.get(endpoint1)

In [132]:
response1

<Response [200]>

In [133]:
query_occurance= pd.DataFrame(response1.json())
query_occurance

month_name  year query_count            query_text month
0         March  2014         101               permits   NaN
1       January  2015         101  criminal court clerk   NaN
2     September  2015         101                 codes   NaN
3         March  2016         101                police   NaN
4         March  2016         101         civil service   NaN
...         ...   ...         ...                   ...   ...
1257  September  2018         750     annual enrollment   NaN
1258    October  2018         816     annual enrollment   NaN
1259    January  2019        2646             Nashville     1
1260  September  2019        5327             directory     9
1261  September  2019        5610                  test     9

[1262 rows x 5 columns]

In [134]:
result = response1.json()
len(result)

1262

How many times did this occur?
query occurance over 1262 times.

In [ ]:
Make another API request that returns all the times "codes" was searched more than 100 times in a month. 
How many times did this occur?

In [30]:
endpoint = 'https://data.nashville.gov/resource/fuaa-r5cm.json?$where=query_count>100'

In [31]:
params = {'query_text': 'codes'}

endpoint = 'https://data.nashville.gov/resource/fuaa-r5cm.json?$where=query_count > 100'

In [32]:
response2 = requests.get(endpoint,params=params)

In [33]:
response2

<Response [200]>

In [34]:
result = response2.json()

In [36]:

result

[{'month_name': 'September',
  'year': '2015',
  'query_count': '101',
  'query_text': 'codes'},
 {'month_name': 'April',
  'month': '4',
  'year': '2021',
  'query_count': '102',
  'query_text': 'codes'},
 {'month_name': 'August',
  'year': '2016',
  'query_count': '104',
  'query_text': 'codes'},
 {'month_name': 'November',
  'month': '11',
  'year': '2020',
  'query_count': '104',
  'query_text': 'codes'},
 {'month_name': 'December',
  'year': '2017',
  'query_count': '106',
  'query_text': 'codes'},
 {'month_name': 'May',
  'year': '2016',
  'query_count': '107',
  'query_text': 'codes'},
 {'month_name': 'April',
  'year': '2016',
  'query_count': '111',
  'query_text': 'codes'},
 {'month_name': 'July',
  'month': '7',
  'year': '2021',
  'query_count': '114',
  'query_text': 'codes'},
 {'month_name': 'January',
  'year': '2017',
  'query_count': '115',
  'query_text': 'codes'},
 {'month_name': 'May',
  'month': '5',
  'year': '2021',
  'query_count': '115',
  'query_text': 'codes'

In [37]:
codes_100 = pd.DataFrame(result)
print (codes_100)

   month_name  year query_count query_text month
0   September  2015         101      codes   NaN
1       April  2021         102      codes     4
2      August  2016         104      codes   NaN
3    November  2020         104      codes    11
4    December  2017         106      codes   NaN
5         May  2016         107      codes   NaN
6       April  2016         111      codes   NaN
7        July  2021         114      codes     7
8     January  2017         115      codes   NaN
9         May  2021         115      codes     5
10     August  2015         117      codes   NaN
11   February  2016         121      codes   NaN
12   February  2019         124      codes     2
13       June  2019         125      codes     6
14  September  2019         126      codes     9
15    January  2019         129      codes     1
16  September  2020         130      codes     9
17   February  2021         131      codes     2
18      March  2019         133      codes     3
19   November  2019 

In [39]:
result = response2.json()
len(result)

56

4. Make an API request that returns the entire Top 500 Monthly Searches dataset. 
    Make a chart that shows the number of times "maps" was searched in a month across the entire time frame.

In [193]:
endpoint = 'https://data.nashville.gov/resource/fuaa-r5cm.json'



In [192]:
params={'query_text':'map'}

In [194]:
response3 = requests.get(endpoint, params = params)

In [195]:
response3

<Response [200]>

In [196]:
result = response3.json()

In [197]:
search_500 = pd.DataFrame(result)
print (search_500)

   month_name  year query_count query_text month
0     January  2014          10        map   NaN
1    February  2014          19        map   NaN
2       March  2014          26        map   NaN
3       April  2014          19        map   NaN
4         May  2014          15        map   NaN
..        ...   ...         ...        ...   ...
77   February  2021          21        map     2
78      March  2021          28        map     3
79      April  2021          25        map     4
80        May  2021          36        map     5
81       July  2021          26        map     7

[82 rows x 5 columns]


search_500['month'] = search_500['month'].astype(str)

 search_500 = search_500.sort_values('query_count', ascending=False,inplace=True)


In [202]:
plt.figure(figsize=(9,6))
plt.bar(search_500['month_name'],search_500['query_count'])
plt.title('Number of Times maps  Searched in a monthh')
# plt.xlable('month_name')
# plt.ylable('query_count')
# pltxticks(rotation=45)
plt.tight_layout()
plt.show()

TypeError: 'NoneType' object is not subscriptable

<Figure size 900x600 with 0 Axes>

In [180]:
endpoint6 = 'https://data.nashville.gov/resource/7qhx-rexh.json'

In [181]:
response6 = requests.get(endpoint6)

In [182]:
response6 

<Response [200]>

In [183]:
result6 = response6.json()

In [184]:
result6

[{'case_number': '966190',
  'status': 'Closed',
  'case_request': 'Resolved by hubNashville on First Call',
  'case_subrequest': 'Resolved by hubNashville on First Call',
  'additional_subrequest': 'Resolved by hubNashville on First Call',
  'date_time_opened': '2022-05-04T17:54:30.000',
  'date_time_closed': '2022-05-04T17:54:30.000',
  'case_origin': 'Phone',
  'state_issue': 'False',
  'closed_when_created': 'True',
  'incident_address': '1301 Pilgrim Dr',
  'incident_city': 'MADISON',
  'incident_council_district': '8',
  'incident_zip_code': '37115',
  'latitude': '36.2718801',
  'longitude': '-86.7175492',
  'mapped_location': {'latitude': '36.2718801', 'longitude': '-86.7175492'},
  ':@computed_region_wvby_4s8j': '1',
  ':@computed_region_3aw5_2wv7': '1',
  ':@computed_region_p6sk_2acq': '3',
  ':@computed_region_gxvr_9jxz': '12',
  ':@computed_region_yf9r_ed6g': '3',
  ':@computed_region_fvtq_wnma': '26',
  ':@computed_region_s8bq_67w7': '3',
  ':@computed_region_v67z_xm3t': '

In [186]:
n_rows=pd.DataFrame(result6)
n_rows.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 33 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   case_number                  1000 non-null   object
 1   status                       1000 non-null   object
 2   case_request                 1000 non-null   object
 3   case_subrequest              1000 non-null   object
 4   additional_subrequest        946 non-null    object
 5   date_time_opened             1000 non-null   object
 6   date_time_closed             995 non-null    object
 7   case_origin                  1000 non-null   object
 8   state_issue                  1000 non-null   object
 9   closed_when_created          1000 non-null   object
 10  incident_address             644 non-null    object
 11  incident_city                637 non-null    object
 12  incident_council_district    637 non-null    object
 13  incident_zip_code            637 n

In [187]:
len(n_rows)

1000